packages/block-editor/src/utils/parse-css-unit-to-px.js

```diff
@@ -117,21 +121,43 @@ function isMathExpression( cssUnit ) {
 function evalMathExpression( cssUnit ) {
 	let errorFound = false;
 	// Convert every part of the expression to px values.
-	const cssUnitsBits = cssUnit
-		.split( /(?!^-)[+*\/-](\s?-)?/g )
-		.filter( Boolean );
-	for ( const unit of cssUnitsBits ) {
+	// The following regex matches numbers that have a following unit
+	// E.g. 5.25rem, 1vw
+	const regex = /\d+\.?\d*[a-zA-Z]+|\.\d+[a-zA-Z]+/g;
+	cssUnit = cssUnit.replace( regex, ( foundUnit ) => {
```

Comment:
> Let me know what you think of [the updated `evalMathExpression`] approach, unit tests pass correctly so I don't expect any regressions, I've tested the font size picker and it's working as expected.

I reviewed https://github.com/WordPress/gutenberg/commit/16155cc8669d020cc032430b1e16e0c5d817938c and the changes make sense to me. 👍🏻 

The only thing that gave me pause was the change from `for` loop to `String.prototype.replace`, namely because the remainder of the function continues to rely upon a `for` loop. It isn't a big deal, it just feels a bit odd to mix the functional and imperative coding paradigms in the same function. 

WDYT? It is fine with me to keep the code as-is. I merely wanted to seek your thoughts.

Reply:
Since it's been a bit since I did this change, I'll review this code and let you know.

Reply:
After reviewing the code I brought back the usage of the for loop in https://github.com/WordPress/gutenberg/pull/47258/commits/391625a9aa681737f9b0636975544279845ba256

It also added a fix for the issue you mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/47258#issuecomment-1438918176). Now we should support nested calc() usages like `clamp(2.625rem, calc(2.625rem + ((1vw - 0.48rem) * 8.4135)), 3.25rem)` which is the current value for the `H2` tag for the Twenty Twenty-Three theme.